### ISSUE TITLE
should define spec for project specific config support

### ISSUE THREAD

**Issue by hoatle** [teracyhq/dev#315]
we should add project specific config support in additional to current organization specific config support

**Comment by hoatle** [comment]
Feature: project specific config

    As a user,
    I want to use different default project specific at a time
    so that I can work with specific projects within the organization at a time

**Comment by hoatle** [comment]
let's take https://github.com/acme101 for the use case.

I want to work on https://github.com/acme101/nextjs-hello-world or https://github.com/acme101/angular-hello-world or other projects at a time, not all projects.

And I want the default config should be kept within the organization level or project levels.

All I need to do:
- clone teracy-dev as acme-dev
- create acme-dev/vagrant_config_override.json with the following content:

```json
{
  "config_paths": [
    "workspace/dev-setup/vagrant_config_default.json",
    "workspace/nextjs-hello-world/vagrant_config_default.json",
    "workspace/angular-hello-world/vagrant_config_default.json"
  ]
}
```
to configure organization level and 2 project specfic levels
=> final vagrant config =  acme-dev/vagrant_config.json < acme-dev/vagrant_config_override.json < workspace/dev-setup/vagrant_config_default.json < workspace/dev-setup/vagrant_config_override.json (if exists) < workspace/nextjs-hello-world/vagrant_config_default.json < workspace/nextjs-hello-world/vagrant_config_override.json (if exists) < workspace/angular-hello-world/vagrant_config_default.json < workspace/angular-hello-world/vagrant_config_override.json (if exists)

or:

```json
{
  "config_paths": [
    "workspace/dev-setup/vagrant_config_default.json",
    "workspace/angular-hello-world/vagrant_config_default.json"
  ]
}
```

=> final vagrant config =  acme-dev/vagrant_config.json < acme-dev/vagrant_config_override.json < workspace/dev-setup/vagrant_config_default.json < workspace/dev-setup/vagrant_config_override.json (if exists) < workspace/angular-hello-world/vagrant_config_default.json < workspace/angular-hello-world/vagrant_config_override.json (if exists)

note that we must keep the override stuff minimal, related: https://github.com/teracyhq/dev/issues/239

**Comment by hoatle** [comment]
spec done, assigned to @phuonglm to review.

**Comment by hoatle** [comment]
auto discovery path for organization config should be implemented, this is similar to #246

By default, load workspace/dev-setup/vagrant_config_default.json if it exists (added to config_paths by default. The configuration path is: vagrant.config_paths.

Users can override this config_paths (of course), see how extension_paths work.

This is used so that users don't have to create any _override.json files if the default always works for them. Users don't have to configure anything if the default works (this works most of the time).

Just git update and the new default config will be updated, users don't have to adjust anything.

Later on, users will know what to do after these kinds of config update, see #318

**Comment by phuonglm** [comment]
#321 done, close
